edb/edgeql/compiler/typegen.py

```diff
@@ -222,9 +222,14 @@ def type_to_typeref(
         expr_type is s_types.ExprType.Update
         or expr_type is s_types.ExprType.Delete
     )
+    include_ancestors = (
+        expr_type is s_types.ExprType.Insert
+        or include_descendants
```

Comment:
It's probably cleaner to restate DML types here in case `include_descendants` deviates from the current set of conditions.